hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/keyvalue/TestTarContainerPacker.java

```diff
@@ -90,17 +93,32 @@ public class TestTarContainerPacker {
   private final ContainerLayoutVersion layout;
   private final String schemaVersion;
   private OzoneConfiguration conf;
+  private String compression;
 
-  public TestTarContainerPacker(ContainerTestVersionInfo versionInfo) {
+
+  public TestTarContainerPacker(
+      ContainerTestVersionInfo versionInfo, String compression) {
     this.layout = versionInfo.getLayout();
     this.schemaVersion = versionInfo.getSchemaVersion();
     this.conf = new OzoneConfiguration();
     ContainerTestVersionInfo.setTestSchemaVersion(schemaVersion, conf);
+    this.compression = compression;
+    packer = new TarContainerPacker(compression);
+
   }
 
   @Parameterized.Parameters
   public static Iterable<Object[]> parameters() {
-    return ContainerTestVersionInfo.versionParameters();
+    List<ContainerTestVersionInfo> layoutList =
+        ContainerTestVersionInfo.getLayoutList();
+    List<Object[]> parameterList = new ArrayList<>();
+    for (ContainerTestVersionInfo containerTestVersionInfo : layoutList) {
+      parameterList.add(new Object[]{containerTestVersionInfo, GZIP});
+      parameterList.add(new Object[]{containerTestVersionInfo, LZ4_FRAMED});
+      parameterList.add(new Object[]{containerTestVersionInfo, SNAPPY_FRAMED});
+      parameterList.add(new Object[]{containerTestVersionInfo, ZSTANDARD});
```

Comment:
Iterate the values of Ozone compression name -> commons-compress name mapping.  Make sure to include `no_compression`.  Reuse `TarContainerPacker.compress/decompress` for handling that case.

Reply:
finished